scripts/test-helpers

```diff
@@ -159,7 +159,12 @@ fetch-kubeconfig() {(
     local num=${1:-1}
     local name=$(cat $TEST_DIR/servers/$num/metadata/name)
     local url=$(cat $TEST_DIR/servers/$num/metadata/url)
-    docker cp $name:/etc/rancher/k3s/k3s.yaml - 2>/dev/null | tar -xO 2>/dev/null | sed -e "s|https://127.0.0.1:6443|$url|g" >$TEST_DIR/servers/$num/kubeconfig.yaml
+    # check for macos and do not replace k8s api url
+    if [[ "$OSTYPE" =~ ^darwin ]]; then
+      docker cp $name:/etc/rancher/k3s/k3s.yaml - 2>/dev/null | tar -xO 2>/dev/null  >$TEST_DIR/servers/$num/kubeconfig.yaml
```

Comment:
I'm curious about this - what is different about OS X that the server URL doesn't need to be updated in order to connect to the port exposed by the container?

Reply:
this does seem like a superfluous conditional. what if youre k3s installation is kind-like aka k3d?

Reply:
There may be a better way to address the issue for running these tests locally on OS X.... The issue on OS X is that docker machine is a VM so the networking gets weird... I added 2 changes here one was to bind 6443 to 6443 on localhost rather than a randomly assigned port. And then not to replace the 127.0.0.1 with what is getting pulled out here:

```
    local ip=$(docker container inspect --format '{{.NetworkSettings.IPAddress}}' $name | tee $TEST_DIR/kine/$count/metadata/ip)
    local port=$(docker container inspect --format '{{range $k, $v := .NetworkSettings.Ports}}{{printf "%s\n" $k}}{{end}}' $name | head -n 1 | cut -d/ -f1 | tee $TEST_DIR/kine/$count/metadata/port)
    local url=$(echo "http://$ip:$port" | tee $TEST_DIR/kine/$count/metadata/url)
```

Without those changes you will get something like this locally on a test run:
```
[nats-2.7.3] Started k3s-server-1-KgHGdM @ https://172.17.0.4:6443
[nats-2.7.3] https://172.17.0.4:6443
[nats-2.7.3] Waiting for kubeconfig to become available...
[nats-2.7.3] https://172.17.0.4:6443
[nats-2.7.3] Running configmap load generation script
[nats-2.7.3] [33353]	W Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x10bb856d0>: Failed to establish a new connection: [Errno 60] Operation timed out')': /version/
[nats-2.7.3] [33350]	W Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x10e625760>: Failed to establish a new connection: [Errno 60] Operation timed out')': /version/
[nats-2.7.3] [33352]	W Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x109f6c5b0>: Failed to establish a new connection: [Errno 60] Operation timed out')': /version/
[nats-2.7.3] [33351]	W Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x111d60070>: Failed to establish a new connection: [Errno 60] Opera
```

The local python k8s client can't connect to url so it fails... But thinking about this now perhaps we can leave the randomly assigned port and connect on `127.0.0.1:<random-port>` but that still might need some special OS X handling.

Reply:
yeah, kind has a port-forwarding config i've had to leverage on os-x and k3s will. iirc, expose a load-balancer on a host-local interface. if we're using docker directly we have to wing it. maybe this is fine for now?